source/main.cc

```diff
@@ -493,48 +493,51 @@ void
 parse_parameters (const std::string &input_as_string,
                   dealii::ParameterHandler  &prm)
 {
-  // try reading on processor 0
-  bool success = true;
-  if (dealii::Utilities::MPI::this_mpi_process (MPI_COMM_WORLD) == 0)
-    try
-      {
-        prm.parse_input_from_string(input_as_string);
-      }
-    catch (const dealii::ExceptionBase &e)
-      {
-        success = false;
-        e.print_info(std::cerr);
-        std::cerr << std::endl;
-      }
-
-
-  // broadcast the result. we'd like to do this with a bool
-  // data type but MPI_C_BOOL is not part of old MPI standards.
-  // so, do the broadcast in integers
-  {
-    int isuccess = (success ? 1 : 0);
-    const int ierr = MPI_Bcast (&isuccess, 1, MPI_INT, 0, MPI_COMM_WORLD);
-    AssertThrowMPI(ierr);
-    success = (isuccess == 1);
-  }
```

Comment:
Here we convert between `bool` and `int`, but we no longer need this as deal.II has learned how to do that conversion itself.